### PR TITLE
doc: add `memory management` group

### PR DESCRIPTION
**Pull request by jia200x**
### Contribution description
This PR does the following:

- Add `sys_memory_management` group
- Move TLSF malloc, oneway malloc and memarray to this new group

### Testing procedure
`make doc`. Go to `System>Memory management` in the generated documentation and see how these components are grouped.

### Issues/PRs references
#8479

### CODE REVIEW

File: sys/oneway-malloc/oneway-malloc.c [hunk 1]
```diff
@@ -7,8 +7,7 @@
  */
 
 /**
- * @addtogroup oneway_malloc
- * @ingroup sys
+ * @ingroup one_way_malloc
```
Comment (jcarrano) [inline sys/oneway-malloc/oneway-malloc.c]: it's oneway, not one_way

Comment (jia200x) [reply sys/oneway-malloc/oneway-malloc.c]: good catch!

**Comment by jcarrano** [comment]
Squash and we're ready to go,

File: sys/oneway-malloc/oneway-malloc.c [hunk 2]
```diff
@@ -7,8 +7,7 @@
  */
 
 /**
- * @addtogroup oneway_malloc
- * @ingroup sys
+ * @ingroup oneway_malloc
```
Comment (smlng) [inline sys/oneway-malloc/oneway-malloc.c]: unrelated to sys_memory_management, or should be:

```
 * @defgroup oneway_malloc
 * @ingroup sys_memory_management
 *
```

Comment (jcarrano) [reply sys/oneway-malloc/oneway-malloc.c]: That will cause `oneway_malloc` to be defined twice, I think.

Comment (smlng) [reply sys/oneway-malloc/oneway-malloc.c]: right, my bad - thought this is the header, but its not - never mind

**Comment by PeterKietzmann** [comment]
@jia200x please address:
```
Commit message is longer than 50 characters:
	    "doc/memory_management: add memoery implementations and utilities to Doxygen group"
```

**Comment by smlng** [comment]
please squash and fix commit message length

**Comment by smlng** [comment]
oops

**Comment by jia200x** [comment]
> @smlng smlng closed this 5 hours ago

:(

**Comment by jia200x** [comment]
addressed!

**Comment by jia200x** [comment]
murdock is happy again :)

**Pull request merged**